### PR TITLE
feat(api): add milestone activity endpoint

### PR DESCRIPTION
**Pull request by Alex6323**

### CODE REVIEW

File: tests/blocks.rs [hunk 1]
```diff
@@ -119,3 +119,67 @@ async fn test_blocks() {
...
+            block,
+            bee_block.pack_to_vec(),
+            BlockMetadata {
+                parents,
+                is_solid: true,
+                should_promote: false,
+                should_reattach: false,
+                referenced_by_milestone_index: 1.into(),
+                milestone_index: 0.into(),
+                inclusion_state: match i {
```
Comment (Alexandcoats) [inline tests/blocks.rs:154]: Not sure there's any point in using anything but `Included` for txns and `NoTransaction` for everything else.

File: src/db/collections/block.rs [hunk 2]
```diff
@@ -247,35 +259,75 @@ impl BlockCollection {
...
+                        "num_treasury_tx_payload": { "$sum": {
+                            "$cond": [ { "$eq": [ "$block.payload.kind", "treasury_transaction" ] }, 1 , 0 ]
+                        } },
+                        "num_milestone_payload": { "$sum": {
+                            "$cond": [ { "$eq": [ "$block.payload.kind", "milestone" ] }, 1 , 0 ]
+                        } },
+                        "num_tagged_data_payload": { "$sum": {
+                            "$cond": [ { "$eq": [ "$block.payload.kind", "tagged_data" ] }, 1 , 0 ]
+                        } },
+                        "num_no_payload": { "$sum": 0 },
```
Comment (grtlr) [inline src/db/collections/block.rs:306]: Reminder for myself

File: tests/blocks.rs (hunk 1 above)
Comment (Alex6323) [reply tests/blocks.rs:154]: huh...not every tx gets included

File: src/db/collections/block.rs (hunk 2 above)
Comment (Alex6323) [reply src/db/collections/block.rs:306]: what does that mean?

File: documentation/api/api-analytics.yml [hunk 3]
```diff
@@ -140,96 +140,47 @@ paths:
...
       responses:
         "200":
-          $ref: "#/components/responses/Block"
+          $ref: "#/components/responses/Address"
         "400":
           $ref: "#/components/responses/BadRequest"
         "500":
           $ref: "#/components/responses/InternalError"
-  /api/analytics/v2/activity/blocks:
+  /api/analytics/v2/activity/milestones/by-id/{milestoneId}:
```
[the remark this answers is not in the document]
Comment (Alex6323) [reply documentation/api/api-analytics.yml:148]: sure

File: src/bin/inx-chronicle/api/stardust/analytics/routes.rs [hunk 4]
```diff
@@ -39,20 +44,14 @@ pub fn routes() -> Router {
...
                     Router::new()
-                        .route("/", get(block_activity_analytics::<()>))
-                        .route("/milestone", get(block_activity_analytics::<MilestonePayload>))
-                        .route("/transaction", get(block_activity_analytics::<TransactionPayload>))
-                        .route("/tagged-data", get(block_activity_analytics::<TaggedDataPayload>))
-                        .route(
-                            "/treasury-transaction",
-                            get(block_activity_analytics::<TreasuryTransactionPayload>),
-                        ),
+                        .route("/by-id/:milestone_id", get(milestone_activity_analytics_by_id))
```
[the remark this answers is not in the document]
Comment (Alex6323) [reply src/bin/inx-chronicle/api/stardust/analytics/routes.rs:50]: sure

File: src/db/collections/block.rs (hunk 2 above)
Comment (Alex6323) [reply src/db/collections/block.rs]: The trick with "$not" that @Alexandcoats suggested works.

Comment (grtlr) [reply src/db/collections/block.rs]: Ah sorry, I didn't want to post that, it was in my pending review (notes).

**Pull request merged**